corehq/apps/sms/views.py

```diff
@@ -1915,6 +1918,30 @@ def dispatch(self, request, *args, **kwargs):
         return super(SMSSettingsView, self).dispatch(request, *args, **kwargs)
 
 
+@method_decorator(toggles.ICDS_CUSTOM_SMS_REPORT.required_decorator(), name='dispatch')
+class SMSUsageReport(BaseMessagingSectionView):
+    template_name = 'sms/usage_report.html'
+    urlname = 'sms_usage_report'
+    page_title = _('SMS Usage Report')
+
+    @use_daterangepicker
+    def dispatch(self, *args, **kwargs):
+        return super(SMSUsageReport, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            start_date, end_date = request.POST.get('date_range').split(' to ')
+            from corehq.messaging.scheduling.forms import validate_date
+            validate_date(start_date)
```

Comment:
it will make it much cleaner if you just add a form class instead and validate there. Its a pretty standard way of doing such validation and also keeps the view cleaner.

Reply:
Can you point me to any specific examples?
I did not quite well understand what do you mean by form class?

Reply:
Check out this form and how the field is validated.
`clean_` methods are called for each field on the form.
https://github.com/dimagi/commcare-hq/blob/42acf2f4f96ad96e1cd0c336441881bf49a0f712/corehq/apps/users/forms.py#L1053

Reply:
Used form classes to create the form and for validation also. I found this approach really `clean` 😁